### PR TITLE
fix(setup): resolve symlinked tools in isolated hook PATH

### LINKED ISSUE #941: bug(setup): no-jq UserPromptSubmit hook drops save nudge on second invocation — test fails on main
### Pre-flight Checks
- [x] I have searched existing issues and this is not a duplicate
- [x] I understand this issue needs status:approved before a PR can be opened

### Bug Description
`TestClaudeCodeUserPromptHookWithoutJQPreservesSessionStateAndNudge` fails on current `main` (3863e98). The no-jq fallback path of the Claude Code UserPromptSubmit hook does not preserve hook state across invocations: the second invocation emits an empty JSON object instead of the expected save nudge. This looks like a regression in the state-preservation logic introduced around #896 (`fix(claude-code): preserve hook state without jq`).

### Steps to Reproduce
1. Checkout main at 3863e98
2. Run: `go test ./internal/setup/ -run TestClaudeCodeUserPromptHookWithoutJQPreservesSessionStateAndNudge -count=1`

### Expected Behavior
Second hook invocation without jq preserves session state and emits the save nudge.

### Actual Behavior
```
setup_test.go:2663: second hook invocation = "{}\n", want save nudge; stderr:
--- FAIL: TestClaudeCodeUserPromptHookWithoutJQPreservesSessionStateAndNudge (0.05s)
```

### Operating System
Linux

### Engram Version
main @ 3863e98 (module path engram/v2)

### Agent / Client
Claude Code

### Relevant Logs
```
FAIL	github.com/Gentleman-Programming/engram/v2/internal/setup	0.059s
```

### Additional Context
Verified failing on both e86ca46 and current origin/main 3863e98, so it predates the last 22 commits. Related open hook issues (#921, #694, #653) cover different failure modes (timeout, unix-socket transport, enforcement no-op) — none covers the no-jq nudge path.

### PR DESCRIPTION
**Pull request by dnlrsls**
## 🔗 Linked Issue

Closes #941

---

## 🏷️ PR Type

- [x] `type:bug` — Bug fix
- [ ] `type:feature` — New feature
- [ ] `type:docs` — Documentation only
- [ ] `type:refactor` — Code refactoring (no behavior change)
- [ ] `type:chore` — Maintenance, dependencies, tooling
- [ ] `type:breaking-change` — Breaking change

---

## 📝 Summary

- Resolve tools found on `PATH` before hardlinking them into the isolated hook test directory.
- Preserve the requested command name when a symlink resolves to a differently named target.
- Add deterministic coverage for regular tools, renamed symlink targets, and broken symlinks.

## 📂 Changes

| File | Change |
|------|--------|
| `internal/setup/setup_test.go` | Resolve symlink targets safely, preserve lookup names, and cover edge cases. |

## 🧪 Test Plan

- [ ] Unit tests pass locally: `go test ./...`
- [ ] E2E tests pass locally: `go test -tags e2e ./internal/server/...`
- [ ] Manually tested the affected functionality

Verification performed on this candidate:

- `go test ./internal/setup -run '^TestExposeHookToolPreservesLookupName$' -count=1` passed.
- `go test ./internal/setup -run '^TestClaudeCodeUserPromptHookWithoutJQPreservesSessionStateAndNudge$' -count=1` passed.
- `go test ./internal/setup -count=1` reached only the known unrelated Windows POSIX-path assertion failures in `TestWriteClaudeCodeUserMCP`.
- `git diff --check` passed.

---

## 🤖 Automated Checks

These run automatically and **all must pass** before merge:

| Check | What it verifies | Status |
|-------|-----------------|--------|
| **Check Issue Reference** | PR body contains `Closes #N` / `Fixes #N` / `Resolves #N` | ⏳ |
| **Check Issue Has status:approved** | Linked issue has `status:approved` label | ⏳ |
| **Check PR Has type:* Label** | PR has exactly one `type:*` label | ⏳ |
| **Unit Tests** | `go test ./...` passes | ⏳ |
| **E2E Tests** | `go test -tags e2e ./internal/server/...` passes | ⏳ |
| **Plugin Tests** | `npm test` passes in `plugin/pi` | ⏳ |

---

## ✅ Contributor Checklist

- [x] I linked an approved issue above (`Closes #N`)
- [x] I added exactly **one** `type:*` label to this PR
- [ ] I ran unit tests locally: `go test ./...`
- [ ] I ran e2e tests locally: `go test -tags e2e ./internal/server/...`
- [x] Docs updated (if behavior changed)
- [x] Commits follow [conventional commits](https://www.conventionalcommits.org/) format
- [x] No `Co-Authored-By` trailers in commits

---

## 💬 Notes for Reviewers

This PR preserves Daniel Gapski's original commit and authorship, then adds a maintainer follow-up for the valid lookup-name edge case identified on #995. It supersedes #995 without changing the production hook.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved hook tool setup for tools installed through symlinks.
  * Preserved the original command name while resolving the underlying tool, including relative symlinks.
  * Prevented dangling links from being created in isolated temporary environments.
  * Improved reliability when tools use alternate names or are located through relative paths.

### CODE REVIEW

File: internal/setup/setup_test.go [hunk 1]
```diff
@@ -3271,6 +3271,139 @@ func TestClaudeCodeUserPromptHookWithoutJQPreservesSessionStateAndNudge(t *testi
...
+			t.Fatalf("read exposed symlinked tool: %v", err)
+		} else if string(data) != "resolved target" {
+			t.Fatalf("exposed symlinked tool = %q, want %q", data, "resolved target")
+		}
+		if _, err := os.Stat(filepath.Join(binDir, "resolved-target")); !os.IsNotExist(err) {
+			t.Fatalf("resolved target basename unexpectedly exposed: %v", err)
+		}
+	})
+
+	t.Run("relative tool path falls back to absolute symlink target", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (dnlrsls) [reply internal/setup/setup_test.go:3325]: Thanks for calling this out. The remaining branch directly propagates the `os.Symlink` error. A missing-directory fixture would primarily test standard-library behavior rather than additional repository logic. Given the test-only scope and existing deterministic fallback-success coverage, we are treating this as non-blocking.

**Comment by danielgap** [comment]
Thanks for this, and for resolving the symlinks in the isolated hook PATH rather than papering over the second-invocation drop.

That closes the exact repro from #941 on my side. Appreciated.
